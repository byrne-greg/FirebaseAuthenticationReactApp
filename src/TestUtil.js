import React from "react";
import { MemoryRouter } from "react-router-dom";

const withTestRouter = Component => <MemoryRouter>{Component}</MemoryRouter>;

export { withTestRouter };
