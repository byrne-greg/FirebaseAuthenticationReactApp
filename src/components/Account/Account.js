import React, { useContext } from "react";
import { AuthUserContext, useAuthorization } from "../Session";
import { PasswordForgetForm } from "../PasswordForget";
import PasswordChangeForm from "../PasswordChange";

const AccountPage = () => {
  const authStrategy = authUser => !!authUser;
  const isAuthorized = useAuthorization(authStrategy);
  const authUser = useContext(AuthUserContext);

  return isAuthorized ? (
    <div>
      <h1>Account: {authUser.email}</h1>
      <PasswordForgetForm />
      <PasswordChangeForm />
    </div>
  ) : null;
};

export default AccountPage;
