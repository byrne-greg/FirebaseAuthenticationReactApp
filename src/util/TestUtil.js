import React from "react";
import { MemoryRouter } from "react-router-dom";
import { FirebaseContext } from "../components/Firebase";
import { AuthUserContext } from "../components/Session";

const withMockFirebase = (mockFirebase, Component) => (
  <FirebaseContext.Provider value={mockFirebase}>
    {Component}
  </FirebaseContext.Provider>
);

const withMockAuthUserContext = (value, Component) => (
  <AuthUserContext.Provider value={value}>{Component}</AuthUserContext.Provider>
);

const withTestRouter = Component => <MemoryRouter>{Component}</MemoryRouter>;

export { withTestRouter, withMockFirebase, withMockAuthUserContext };
