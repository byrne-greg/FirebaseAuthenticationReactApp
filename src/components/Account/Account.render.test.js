import React from "react";
import { render, cleanup } from "@testing-library/react";
import { withTestRouter, withMockAuthUserContext } from "../../util/TestUtil";
import Account from ".";

afterEach(cleanup);

describe("Navigation render tests", () => {
  it("when user is unauthenticated, the public navigation bar renders as expected", () => {
    // GIVEN

    // WHEN
    const page = render(withTestRouter(<Account />));

    // THEN
    expect(page.baseElement).toMatchSnapshot();
  });

  it("when user is authenticated, the authenticated account page renders as expected", () => {
    // GIVEN

    // WHEN
    const page = render(
      withTestRouter(
        withMockAuthUserContext({ email: "test@test.com" }, <Account />)
      )
    );

    // THEN
    expect(page.baseElement).toMatchSnapshot();
  });
});
