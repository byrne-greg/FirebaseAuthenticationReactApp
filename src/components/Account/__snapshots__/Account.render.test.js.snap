// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation render tests when user is authenticated, the authenticated account page renders as expected 1`] = `
<body>
  <div>
    <div>
      <h1>
        Account: 
      </h1>
      <h2>
        Forget your password?
      </h2>
      <form>
        <label
          id="email-label"
        >
          Email Address
        </label>
        <input
          aria-labelledby="email-label"
          id="email-input"
          name="email"
          placeholder="Email Address"
          type="email"
          value=""
        />
        <button
          disabled=""
          id="submit-button"
          type="submit"
        >
          Reset My Password
        </button>
      </form>
      <div>
        <h1>
          Change Password
        </h1>
        <h2>
          Change your password
        </h2>
        <form>
          <label
            id="new-password-label"
          >
            New Password
          </label>
          <input
            aria-labelledby="new-password-label"
            name="new-password"
            placeholder="New Password"
            type="password"
            value=""
          />
          <label
            id="confirm-new-password-label"
          >
            Confirm New Password
          </label>
          <input
            aria-labelledby="confirm-new-password-label"
            name="confirm-new-password"
            placeholder="Confirm New Password"
            type="password"
            value=""
          />
          <button
            disabled=""
            id="submit-button"
            type="submit"
          >
            Change Password
          </button>
        </form>
      </div>
    </div>
  </div>
</body>
`;

exports[`Navigation render tests when user is unauthenticated, the public navigation bar renders as expected 1`] = `
<body>
  <div>
    <div>
      <h1>
        You are not allowed to access this page
      </h1>
    </div>
  </div>
</body>
`;
